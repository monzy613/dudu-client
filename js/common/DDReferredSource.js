/**
 * @providesModule DDReferredSource
 */
import React, { Component } from 'react'
import {
  View,
  Text,
  TouchableOpacity,
  StyleSheet,
} from 'react-native'

import {
  backgroundColor,
  darkText,
  lightText,
} from 'DDColor'

export default class DDReferredSource extends Component {
  render = () => {
    const {
      feed = {},
      onPress,
      style = {},
    } = this.props
    const {
      title,
      source,
    } = feed
    return (
      <TouchableOpacity style={[styles.container, style]} onPress={onPress} activeOpacity={1}>
        <Text style={styles.title} numberOfLines={1}>{title}</Text>
        <Text style={styles.source} numberOfLines={1}>{source}</Text>
      </TouchableOpacity>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: 'white',
    borderRadius: 2,
    borderWidth: 1,
    borderColor: backgroundColor,
    justifyContent: 'space-between',
    padding: 10,
  },
  title: {
    color: darkText,
    fontSize: 16,
    fontWeight: '100',
  },
  source: {
    color: lightText,
    fontWeight: '100',
    marginTop: 5,
  },
})
