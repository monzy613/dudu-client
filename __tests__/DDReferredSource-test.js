import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'

import DDReferredSource from 'DDReferredSource'

describe('common components', () => {
  it('DDReferredSource renders correctly', () => {
    const tree = renderer.create(<DDReferredSource />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})