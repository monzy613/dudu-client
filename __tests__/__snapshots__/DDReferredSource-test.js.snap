exports[`common components DDReferredSource renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "white",
      "borderColor": "#F2F2F2",
      "borderRadius": 2,
      "borderWidth": 1,
      "justifyContent": "space-between",
      "opacity": 1,
      "padding": 10,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    numberOfLines={1}
    style={
      Object {
        "color": "#000000",
        "fontSize": 16,
        "fontWeight": "100",
      }
    } />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    numberOfLines={1}
    style={
      Object {
        "color": "#9B9B9B",
        "fontWeight": "100",
        "marginTop": 5,
      }
    } />
</View>
`;
